### PR TITLE
fix(radio-button): margin bug fix

### LINKED ISSUE #7113: Radio button's focus state is truncated when in a container with a hidden overflow
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.
When a radio button is used inside a container with a hidden or auto overflow set, the box-shadow used to show the focused state of the radio button is truncated

> Is this issue related to a specific component?
Radio button

> What did you expect to happen? What happened instead? What would you like to
> see changed?
Provide some padding around the control to account for the focus state's box-shadow

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
latest 

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1. Go to https://codesandbox.io/s/damp-grass-kjx1w?file=/index.js
2. Select the radio button to show a focused state
3. Observe that the box-shadow around the radio button is cut off below and on the left

> Please create a reduced test case in CodeSandbox
>https://codesandbox.io/s/damp-grass-kjx1w?file=/index.js

## Additional information

- Screenshots or code
image

- Notes
A similar issue was recently raised and addressed with Checkbox here: https://github.com/carbon-design-system/carbon/issues/6480

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #7113 

added a margin around radio button to fix truncation bug

#### Changelog

**New**

- `0.25 rem to margin-top & margin-bottom` around radio-button

#### Testing / Reviewing
 

- [ ] Go to: https://deploy-preview-7133--carbon-components-react.netlify.app/?path=/story/radiobutton--default

- [ ] add border & overflow around radio-button `bx--radio-button__label` as seen below `border: 10px solid red;` `overflow: hidden;`

- [ ] make sure radio-button's focus state is not cut off / truncated

Screen Shot 2020-10-26 at 11 46 41 AM

### CODE REVIEW

**Comment by joshblack** [comment]
@andreancardona if I understand this issue, it's that the box-shadow bleeds over the box model of the radio button? Just wanted to confirm 👀

**Comment by andreancardona** [comment]
> @andreancardona if I understand this issue, it's that the box-shadow bleeds over the box model of the radio button? Just wanted to confirm 👀

Yep! Let me know if there is a better way to solve for it

**Comment by joshblack** [comment]
@andreancardona this makes me think about the focus style itself, I'm curious if the shadow should be inset then instead? Would require a sync with design but it seems like our focus styles should probably stay scoped inside of the container instead of growing otherwise we do have to add in some extra spacing to accommodate for it 🤔

**Comment by andreancardona** [comment]
> kes me think about the focus style itself, I'm curious if the shadow should be inset then instead? Would require a sync with design but it seems like our focus styles should probably stay scoped inside of the contai

Yeah that makes a lot more sense. What is the best next step then in this situation?

File: packages/components/src/components/radio-button/_radio-button.scss [hunk 1]
```diff
@@ -71,7 +71,7 @@
     flex-shrink: 0;
     width: rem(18px);
     height: rem(18px);
-    margin-right: $carbon--spacing-03;
+    margin: 0.2rem $carbon--spacing-03 0.2rem 0;
```
Comment (joshblack) [inline packages/components/src/components/radio-button/_radio-button.scss]: Would this make sense as:

```suggestion
    margin: 2px $carbon--spacing-03 2px 0;
```

Which would account for the outline size? Or is the 3.2px value exact? Wasn't sure where it was coming from 👀

Comment (andreancardona) [reply packages/components/src/components/radio-button/_radio-button.scss]: @joshblack that makes sense! I guess I just thought we always want to use `rem` - just in general what is the rule of thumb around that? Re: when to use `rem` v `px`?

Comment (joshblack) [reply packages/components/src/components/radio-button/_radio-button.scss]: @andreancardona let me know if I'm missing something, but it seemed like this margin was being used to offset the outline width which was in px. If so, and we used rem, then we might go past the size of the outline if the font-size increases even though the outline width itself didn't change.

**Review by joshblack — APPROVED** [review]

**Comment by joshblack** [comment]
bump @dakahn if you have a sec to review today

**Comment by tw15egan** [comment]
Just curious, if we're adding a margin to the top and bottom to account for the focus, should we also add some to the left side?

Screen Shot 2020-11-05 at 11 47 52 AM

**Review by dakahn — APPROVED** [review]
seems cool to me, pending @tw15egan 's questions

**Comment by andreancardona** [comment]
> Just curious, if we're adding a margin to the top and bottom to account for the focus, should we also add some to the left side?
> 
> Screen Shot 2020-11-05 at 11 47 52 AM

good catch! updated!

**Pull request merged**